### PR TITLE
[FLINK-13347] [table-planner] should handle SEMI/ANTI JoinRelType in switch case

### PR DESCRIPTION
**Pull request by godfreyhe**
## What is the purpose of the change

*should handle SEMI/ANTI JoinRelType in switch case*

## Brief change log

  - *handle SEMI/ANTI JoinRelType in switch case*

## Verifying this change

all tests should pass

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not **documented)**

### CODE REVIEW

**Comment by godfreyhe** [comment]
cc @walterddr

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamJoinToCoProcessTranslator.scala [hunk 1]
```diff
@@ -145,6 +145,7 @@ class DataStreamJoinToCoProcessTranslator(
           genFunction.name,
           genFunction.code,
           queryConfig)
+      case _ => throw new TableException(s"$joinType is not supported.")
```
Comment (walterddr) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamJoinToCoProcessTranslator.scala]: I think this one should throw `ValidationException` similar to other Translator classes.

**Comment by godfreyhe** [comment]
thanks for your suggestion @walterddr,  i have updated

**Comment by walterddr** [comment]
@flinkbot approve all

**Pull request closed without merging**